src/Database/PostgreSQL/PQTypes/Model/Index.hs

```diff
@@ -36,6 +41,32 @@ data TableIndex = TableIndex {
 , idxWhere   :: Maybe (RawSQL ())
 } deriving (Eq, Ord, Show)
 
+data IndexColumn
+  = IndexColumn (RawSQL ()) (Maybe (RawSQL ()))
+  deriving Show
+
+-- If one of the two columns doesn't specify the operator class, we just ignore
+-- it and still treat them as equivalent.
+instance Eq IndexColumn where
```

Comment:
Can you check if the Eq instance is needed?

Reply:
It's used in `checkIndexes` which uses `checkEquality`.